moveit_ros/occupancy_map_monitor/include/moveit/occupancy_map_monitor/occupancy_map_monitor.h

```diff
@@ -36,30 +36,80 @@
 
 #pragma once
 
-#include <vector>
+#include <functional>
+#include <memory>
+#include <mutex>
 #include <string>
+#include <vector>
+
 #include <rclcpp/rclcpp.hpp>
 #include <tf2_ros/buffer.h>
-#include <pluginlib/class_loader.hpp>
 
 #include <moveit_msgs/srv/save_map.hpp>
 #include <moveit_msgs/srv/load_map.hpp>
 #include <moveit/occupancy_map_monitor/occupancy_map.h>
 #include <moveit/occupancy_map_monitor/occupancy_map_updater.h>
 
-#include <boost/thread/mutex.hpp>
-
-#include <memory>
-
 namespace occupancy_map_monitor
 {
 class OccupancyMapMonitor
 {
 public:
+  /**
+   * @brief      This class contains the rcl interfaces for easier testing
+   */
+  class MiddlewareHandle
+  {
+  public:
+    using SaveMapServiceCallback = std::function<bool(const std::shared_ptr<rmw_request_id_t> request_header,
+                                                      const std::shared_ptr<moveit_msgs::srv::SaveMap::Request> request,
+                                                      std::shared_ptr<moveit_msgs::srv::SaveMap::Response> response)>;
+    using LoadMapServiceCallback = std::function<bool(const std::shared_ptr<rmw_request_id_t> request_header,
+                                                      const std::shared_ptr<moveit_msgs::srv::LoadMap::Request> request,
+                                                      std::shared_ptr<moveit_msgs::srv::LoadMap::Response> response)>;
+
+    virtual ~MiddlewareHandle() = default;
+    virtual double getMapResolutionParameter() const = 0;
+    virtual std::string getMapFrameParameter() const = 0;
+    virtual std::vector<std::string> getSensorListParameter() const = 0;
+    virtual std::string getSensorPluginParameter(const std::string& sensor) = 0;
```

Comment:
Where did the documentation go? Also, this couldn't be made `const`? Why am I not surprised? Is it because of the node handle? Would it make sense to make it `mutable` so that anything else would have `const` enforcement?

Reply:
It is because it has to call get_parameter... I could move more logic into the constructor of the handle but I was trying to keep it as logic free as possible.

Reply:
Also, I just noticed, that in the ctor of `OccupancyMapMonitor`, `getSensorListParameter` is called, and then immediately passed to `getSensorPluginParameter`, and isn't stored. Why is it better to have this logic in the `OccupancyMapMonitor` ctor, rather than the `MiddlewareHandle`, which also has logic. You could also then remove `getSensorListParameter` from the public interface.

Reply:
done, lmk what you think of my implementation of this.